hudi-common/src/main/java/org/apache/hudi/common/bloom/HoodieDynamicBoundedBloomFilter.java

```diff
@@ -125,5 +126,42 @@ private void extractAndSetInternalBloomFilter(DataInputStream dis) throws IOExce
     internalDynamicBloomFilter = new InternalDynamicBloomFilter();
     internalDynamicBloomFilter.readFields(dis);
   }
+
+  @Override
+  public void or(BloomFilter other) {
+    if (other != null) {
+      ValidationUtils.checkArgument(other instanceof HoodieDynamicBoundedBloomFilter,
+          "HoodieDynamicBoundedBloomFilter can only perform OR operations with other HoodieDynamicBoundedBloomFilter.");
+      HoodieDynamicBoundedBloomFilter otherFilter = (HoodieDynamicBoundedBloomFilter) other;
+      int sourceMatrixLength = this.getMatrixLength();
+      int targetMatrixLength = otherFilter.getMatrixLength();
+      if (targetMatrixLength > sourceMatrixLength) {
+        this.rescaleFromTarget(targetMatrixLength);
+      } else {
+        otherFilter.rescaleFromTarget(sourceMatrixLength);
+      }
+      this.internalDynamicBloomFilter.or(otherFilter.internalDynamicBloomFilter);
+    }
+  }
+
+  /**
+   * rescale the internal dynamic bloom filter by length
+   * @param targetMatrixLength
+   * @return
+   */
+  private HoodieDynamicBoundedBloomFilter rescaleFromTarget(int targetMatrixLength) {
+    int initMatrixLength = this.internalDynamicBloomFilter.getMatrixLength();
+    int needAddRowNum = targetMatrixLength - initMatrixLength;
+    if (needAddRowNum > 0) {
+      for (int i = 0; i < needAddRowNum; i++) {
+        this.internalDynamicBloomFilter.addRow();
```

Comment:
Not sure if we can rebuild the filter in one shot instead of invoking the `addRow` multiple times?